### PR TITLE
Add test for policies from private registries

### PR DESCRIPTION
**Pull request by kravciak**
Add test for installation of policy from registry with authentication.

Requested in https://github.com/kubewarden/kubewarden-controller/pull/421#issuecomment-1491718639
Blocked on: https://github.com/kubewarden/kwctl/issues/476

### CODE REVIEW

File: tests/private-registry-tests.bats [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+@test "[Private Registry] Pull & Push policy to registry" {
+	config="$BATS_RUN_TMPDIR/config.json"
+	jq -n --arg r $REGISTRY '{"auths": {($r): {"auth": "dGVzdHVzZXI6dGVzdHBhc3N3b3Jk"}}}' > $config
+
+	kwctl pull $PUB_POLICY
+	kwctl push $PUB_POLICY $PRIV_POLICY \
+		--docker-config-json-path $BATS_RUN_TMPDIR \
+		--sources-path <(echo "insecure_sources: [$REGISTRY]")
```
Comment (flavio) [inline tests/private-registry-tests.bats:57]: if you want to go the extra mile, you could create a `sources.json` file that holds the certificate you created before, something like that

```yml
source_authorities:
  "registry-testing.svc.lan":
  - type: Path
    path: /etc/pki/trust/anchors/home.pem
```

File: tests/private-registry-tests.bats [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	# Create secret to access registry
+	kubectl --namespace kubewarden create secret docker-registry secret-registry-docker \
+	  --docker-username=testuser \
+	  --docker-password=testpassword \
+	  --docker-server=$REGISTRY
+
+	# Edit default policy server config
+	helm_in kubewarden-defaults --reuse-values \
+		--set policyServer.imagePullSecret=secret-registry-docker \
+		--set policyServer.insecureSources[0]=$REGISTRY
```
Comment (flavio) [inline tests/private-registry-tests.bats:71]: In this case you should set the `sourceAuthorities` section of the PolicyServer deployment. The key is the name of the registry, the value is the PEM encoded cert

**Review by flavio — APPROVED** [review]
LGTM, I left some minor comments. These comments would allow us to test other aspects of the private registries (the usage of custom CA)

**Comment by kravciak** [comment]
Rust was failing when I used self-signed certificates with `invalid peer certificate contents: invalid peer certificate: CaUsedAsEndEntity`, so I had to rework certificate generation.

It should work according to comments, can you review pls?

**Pull request merged**